gcloud-java-core/src/main/java/com/google/gcloud/BaseServiceException.java

```diff
@@ -48,10 +48,16 @@
 
     private final Integer code;
     private final String reason;
+    private final boolean rejected;
 
     public Error(Integer code, String reason) {
+      this(code, reason, false);
+    }
+
+    public Error(Integer code, String reason, boolean rejected) {
```

Comment:
We probably should for completeness, however setting up rejected only matters if the operation is not idempotent.
When idempotent is false we will not retry even if there is a match with a retryable exception unless we know for
sure that the server didn't process the exception (`rejected == true`).

In Storage I think all the exception are thrown as `idempotent == true`, so technically this will not change much there.